frontend/src/v5/validation/auth.ts

```diff
@@ -16,11 +16,41 @@
  */
 
 import * as Yup from 'yup';
+import { formatMessage } from '../services/intl';
 
-const username = Yup.string().required();
+export const username = Yup.string().required();
 const password = Yup.string().required();
 
 export const LoginSchema = Yup.object().shape({
 	password,
 	username,
 });
+
+export const PasswordForgotSchema = Yup.object().shape({
+	username,
+});
+
+export const PasswordChangeSchema = Yup.object().shape({
+	newPassword: Yup.string()
+		.required(
+			formatMessage({
+				id: 'passwordChange.error.passwordRequired',
+				defaultMessage: 'Password is required',
+			}),
+		)
+		.max(65,
+			formatMessage({
+				id: 'passwordChange.error.tooLong',
+				defaultMessage: 'Password is limited to 65 characters',
+			})),
+	newPasswordConfirm: Yup.string()
+		.required(formatMessage({
+			id: 'passwordChange.error.passwordRequired',
+			defaultMessage: 'Password is required',
+		}))
+		.oneOf([Yup.ref('newPassword')],
+			formatMessage({
+				id: 'passwordChange.error.notMatching',
+				defaultMessage: 'The two entered passwords do not match',
```

Comment:
In the signup form, I wrote this error message for when the passwords do not match: "Passwords must match". Maybe we could ask George what to put there, and we both use what he says so that we keep it consinstent.